<!--
asap_primary_LLP.launch

Keenan Albee and Charles Oestreich, 2021
MIT Space Systems Laboratory
-->

<launch>
  <!-- Debug options-->
  <?comment launch-prefix="x-terminal-emulator -e gdb -ex run --args" ?>
  <?comment launch-prefix="gdb -ex run --args" ?>
  <?comment launch-prefix="valgrind --leak-check=full" ?>

  <!-- Additional options -->
  <arg name="drivers"/>                       <!-- Should we launch drivers? -->
  <arg name="spurn" default=""/>              <!-- PRevent a specific node   -->
  <arg name="nodes" default=""/>              <!-- Launch specific nodes     -->
  <arg name="extra" default=""/>              <!-- Inject an additional node -->
  <arg name="debug" default=""/>                 <!-- Debug a node set          -->
  <arg name="output" default="log"/>          <!-- Where to log              -->
  <arg name="ns"    default="" />             <!-- Robot namespace           -->
  <arg name="loc_only" default="false"/>           <!-- Only run localization     -->
  <arg name="ground" default="false" />          <!-- Is this a ground test? (optional use) -->
  <arg name="sim" default="false" />             <!-- Is this a sim test? (optional use)    -->

  <rosparam command="load" file="$(find execute_asap)/config/casadi_nmpc_params.yaml"/>

  <!-- Uses an unfortunate hack to suppress all verbose QPOASES output-->
  <group ns="$(arg ns)">
    <node pkg="nodelet" type="nodelet" name="casadi_nmpc"
          required="false" respawn="false"
          args="standalone casadi_nmpc/CasadiNMPCNodelet"
          output="log"/>
  </group>
</launch>
